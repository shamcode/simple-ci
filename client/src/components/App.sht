{% import ActivePageContainer from 'sham-ui-router/active-page-container' %}
{% import Header from './Header.sht' %}
{% import Footer from './Footer.sht' %}

<Header/>
<main>
    <ActivePageContainer/>
</main>
<Footer
    version={{version}}
/>
