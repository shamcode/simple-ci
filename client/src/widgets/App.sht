{% import Header from './Header.sht' %}
{% import Footer from './Footer.sht' %}

<Header/>
<main>
    <pre>
        Client fo simple-ci
    </pre>
</main>
<Footer/>
