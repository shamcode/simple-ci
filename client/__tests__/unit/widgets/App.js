import { DI } from 'sham-ui';
import App from '../../../src/widgets/App.sht';
import renderer from 'sham-ui-test-helpers';
import DummyPage from './DummyPage.sht';

afterEach( () => {
    DI.bind( 'router', null );
} );

it( 'renders correctly', () => {
    DI.bind( 'router', {
        generate: jest.fn().mockReturnValueOnce( '/' ),
        activePageWidget: DummyPage,
        lastRouteResolved: jest.fn().mockReturnValueOnce( '/' )
    } );

    const meta = renderer( App, {} );
    expect( meta.toJSON() ).toMatchSnapshot();
} );
