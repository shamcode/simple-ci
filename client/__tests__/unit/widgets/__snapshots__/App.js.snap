// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "Constructor": "App",
  "Options": Object {
    "ID": "widget",
    "containerSelector": "body",
    "types": Array [],
  },
  "html": "
<header>
  <a href=\\"/\\" class=\\"logo\\">Simple CI
    <!--default-->
  </a>
</header>
<!--Header-->
<main></main>
<footer> Sham 2018 </footer>
<!--Footer-->
",
}
`;
